### PR TITLE
chore: remove duplicated code in ReadOperation.ts

### PR DESCRIPTION
**Pull request by Andy2003**
# Description

This commit harmonizes the logic of `transpileNestedRelationship` with the one of `transpile`

## Complexity

Complexity: low

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [x] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 43 | 47 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 43 | 77 | 1134 |
|  aggregations.NestedAggregation | 16544 | 16544 | 78 | 83 | 2174 |
|  aggregations.AggregationWithWhere | 11970 | 11970 | 38 | 53 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116478 | 22116478 | 2160 | 2418 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116478 | 22116478 | 2231 | 2194 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9701 | 9701 | 30 | 44 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9928 | 9928 | 32 | 40 | 2174 |
|  aggregations.DeeplyNestedCount | 11062184 | 11062184 | 2928 | 2542 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 44 | 65 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 72 | 121 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 117 | 164 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 96 | 64 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 186 | 183 | 3003 |
|  connections.Connection | 14049 | 14049 | 59 | 65 | 2174 |
|  connections.NestedConnection | 41068 | 41068 | 117 | 139 | 4516 |
|  create.SimpleMutation | 7 | 7 | 55 | 79 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 66 | 25 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 224 | 217 | 1040 |
|  delete.NestedDeleteInUpdate | 19165 | 19165 | 185 | 199 | 1183 |
|  2871.NestedRelationshipFilter | 23081 | 23103 | 69 | 62 | 4436 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 40 | 51 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22891 | 22891 | 74 | 114 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 33 | 52 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 48 | 60 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 19 | 25 | 1040 |
|  query.SimpleQueryWithRelationship | 16129 | 16129 | 40 | 45 | 2174 |
|  query.QueryWhere | 9684 | 9679 | 32 | 39 | 2164 |
|  query.SimpleQueryWithNestedWhere | 9851 | 9846 | 54 | 54 | 2164 |
|  query.Nested | 10095502 | 10095502 | 7170 | 7141 | 2008534 |
|  query.NestedWithFilter | 10074351 | 10074351 | 6694 | 6846 | 2004000 |
|  query.OrFilterOnRelationships | 42068 | 41198 | 153 | 206 | 1635 |
|  query.OrFilterOnRelationshipsAndNested | 35764 | 35311 | 178 | 282 | 1635 |
|  query.QueryWithNestedIn | 13331 | 13323 | 49 | 80 | 1205 |
|  query.NestedConnectionWhere | 9801 | 9801 | 52 | 69 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9860 | 9930 | 87 | 90 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18800 | 18776 | 116 | 169 | 1598 |
|  query.NestedWithRelationshipSingleFilters | 3865 | 3865 | 137 | 187 | 1134 |
|  query.Fulltext | 64 | 64 | 28 | 38 | 16 |
|  query.FulltextWithNestedQuery | 558 | 558 | 40 | 48 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14059 | 14059 | 44 | 49 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14059 | 14059 | 65 | 75 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14155 | 14169 | 57 | 65 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14155 | 14169 | 112 | 112 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14592 | 14651 | 101 | 132 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14437 | 14511 | 99 | 127 | 2174 |
|  sorting.SortMultipleTypes | 3498 | 3498 | 86 | 99 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14532 | 14462 | 113 | 117 | 2174 |
|  sorting.SortOnNestedFields | 14049 | 14049 | 53 | 50 | 2174 |
|  sorting.SortDeeplyNestedFields | 43148 | 43148 | 76 | 83 | 4516 |
|  sorting.ConnectionWithSort | 3281 | 3281 | 75 | 81 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 60 | 79 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 57 | 57 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 288 | 253 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 229 | 241 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 112 | 105 | 2002 |

</details>
        
Old Schema Generation: 24.383s
Schema Generation: 24.795s
Old Subgraph Schema Generation: 28.552s
Subgraph Schema Generation: 28.354s

File: packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ import { CypherPropertySort } from "../sort/CypherPropertySort";
 import type { Sort } from "../sort/Sort";
 import type { OperationTranspileResult } from "./operations";
 import { Operation } from "./operations";
+import type { CompositeClause } from "@neo4j/cypher-builder/dist/clauses/utils/concat";
```
Comment (MacondoExpress) [inline packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts]:
```suggestion
```

File: packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts [hunk 2]
```diff
@@ -199,101 +127,95 @@ export class ReadOperation extends Operation {
...
-        if (wherePredicate) {
-            if (filterSubqueryWith) {
-                filterSubqueryWith.where(wherePredicate); // TODO: should this only be for aggregation filters?
-            } else if (extraMatches.length) {
-                extraMatchesWith = new Cypher.With("*");
-                extraMatchesWith.where(wherePredicate);
-            } else {
-                matchClause.where(wherePredicate);
+        let sortAndLimitBlock: Cypher.Clause | undefined = undefined;
+        let subqueries: CompositeClause;
```
Comment (MacondoExpress) [inline packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts]:
```suggestion
        let subqueries: Cypher.Clause;
```

**Review by MacondoExpress — CHANGES_REQUESTED** [review]
Thanks a lot for this effort @Andy2003! I left two small changes to make, but apart from that looks good!

File: packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts [hunk 3]
```diff
@@ -199,101 +126,95 @@ export class ReadOperation extends Operation {
...
-        });
+        const ret: Cypher.Return = this.relationship
+            ? this.getProjectionClause(nestedContext, context.returnVariable, this.relationship.isList)
+            : this.getReturnStatement(
+                  isCreateSelection || isUpdateSelection ? context : nestedContext,
+                  nestedContext.returnVariable
+              );
 
         let filterSubqueryWith: Cypher.With | undefined;
         let filterSubqueriesClause: Cypher.Clause | undefined = undefined;
```
Comment (MacondoExpress) [inline packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts]:
```suggestion
        let filterSubqueriesClause: Cypher.Clause | undefined;
```

File: packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts [hunk 4]
```diff
@@ -199,101 +126,95 @@ export class ReadOperation extends Operation {
...
-        let extraMatchesWith: Cypher.With | undefined;
-        if (wherePredicate) {
-            if (filterSubqueryWith) {
-                filterSubqueryWith.where(wherePredicate); // TODO: should this only be for aggregation filters?
-            } else if (extraMatches.length) {
-                extraMatchesWith = new Cypher.With("*");
-                extraMatchesWith.where(wherePredicate);
-            } else {
-                matchClause.where(wherePredicate);
+        let sortAndLimitBlock: Cypher.Clause | undefined = undefined;
```
Comment (MacondoExpress) [inline packages/graphql/src/translate/queryAST/ast/operations/ReadOperation.ts]:
```suggestion
        let sortAndLimitBlock: Cypher.Clause | undefined;
```

**Review by MacondoExpress — CHANGES_REQUESTED** [review]

**Review by MacondoExpress — APPROVED** [review]

**Pull request merged**